packages/notebook-extension/src/index.ts

```diff
@@ -974,14 +974,59 @@ function addCommands(
         const end = editor.getOffsetAt(selection.end);
         code = editor.model.value.text.substring(start, end);
       } else {
-        // no selection, submit whole line and advance
-        code = editor.getLine(selection.start.line);
+        // no selection, find the complete statement around the current line
         const cursor = editor.getCursorPosition();
-        if (cursor.line + 1 !== editor.lineCount) {
-          editor.setCursorPosition({
-            line: cursor.line + 1,
-            column: cursor.column
-          });
+        let srcLines = editor.model.value.text.split('\n');
```

Comment:
@jasongrout Should I also consider `\r\n` stuff here?

Reply:
I think the editor automatically converts \r\n to \n, so you can consistently use \n. @ian-r-rose?

Reply:
In the notebook we never use `\r\n`, so I don't think it should ever be in there unless you use some alternative notebook editing software.

We make an effort to keep convert `\r\n` to `\n` when editing text files, and then restore them upon saving.